crates/json-abi/src/param.rs

```diff
@@ -178,20 +400,24 @@ impl EventParam {
         }
     }
 
+    fn borrowed_internal_type(&self) -> Option<BorrowedInternalType<'_>> {
+        self.internal_type().as_ref().map(|it| it.as_borrowed())
+    }
+
     #[inline]
     fn as_inner(&self) -> BorrowedParam<'_, Param> {
         BorrowedParam {
             name: &self.name,
             ty: &self.ty,
             indexed: Some(self.indexed),
-            internal_type: self.internal_type.as_deref(),
+            internal_type: self.borrowed_internal_type(),
             components: Cow::Borrowed(&self.components),
         }
     }
 }
 
-#[derive(Deserialize, Serialize)]
-#[serde(bound(deserialize = "<[T] as ToOwned>::Owned: Default + Deserialize<'de>"))]
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(bound(deserialize = "<[T] as ToOwned>::Owned: Default + Deserialize<'de>, 'a: 'de"))]
```

Comment:
lifetime bound is added by serde by doing `#[serde(borrow)]`, i dont think you need to add it here too

Reply:
it refused to compile without. I am not familiar enough with the serde internals to know why yet